### ISSUE TITLE
OpenAI API new Edit mode

### ISSUE THREAD

**Issue by djack42** [jeffdapaz/VisualChatGPTStudio#21]
Hi, [OpenAI released new modes](https://openai.com/blog/gpt-3-edit-insert) to query the API : [Edit](https://platform.openai.com/docs/api-reference/edits) and Insert, in addition to the Chat mode.  
I used Playground to try them and I think it would produce better results than the chat mode for this extension, at least for refactor commands :  
![image](https://user-images.githubusercontent.com/9913996/226574724-809dfa14-2560-48a2-8a6e-2797a957639f.png)  

I see that the extension is using [OpenAI](https://www.nuget.org/packages/OpenAI) byOkGoDoIt V1.6.0, but it does not have these modes yet. But a fork exists : [OpenAI-DotNet](https://www.nuget.org/packages/OpenAI-DotNet/), more up to date with the API, including the Edit mode. It's the same init, that could be intersting to try it out.  

Anyway, big thanks for your hard work!

**Comment by jeffdapaz** [comment]
To be honest I've noticed these models before but hadn't read about them. But now that you mention it, I've read the documentation and it looks interesting. It would have made it a little easier at the beginning of the development of the extension, as I wouldn't have to do workarounds depending on the command so that the response is coherent and also that it is correctly positioned in the code.

However, I believe that to make these models available for the extension's commands, it would be necessary for the user to choose for each command which model he would like to use, because depending on the command, one would be better than the other.

As for the tool window, perhaps a combo with the listing of models in the window itself could provide users with freedom and ease to choose the best way to make requests, as well as allowing them to obtain the best possible results.

I also saw this other library, it seems to be much more complete and updated than the extension being used at the moment. At the moment, if someone implement these new models, would have to make the effort to change the library used.

Finally, thank you for your contributions with new ideas.
